### PR TITLE
ts-web: trying out other crypto libraries

### PR DESCRIPTION
**Pull request by pro-wh**
cross reference #1371

### CODE REVIEW

**Comment by CedarMist** [comment]
I second this pull request, the `@noble/` ECC related libraries are much cleaner than `elliptic`.

File: client-sdk/ts-web/rt/src/signature_secp256k1.ts [hunk 1]
```diff
@@ -44,28 +38,28 @@ export class BlindContextSigner implements ContextSigner {
     }
 }
 
-export class EllipticSigner implements Signer {
-    key: elliptic.ec.KeyPair;
+export class NobleSigner implements Signer {
+    key: Uint8Array;
 
-    constructor(key: elliptic.ec.KeyPair, note: string) {
+    constructor(key: Uint8Array, note:string) {
```
Comment (pro-wh) [inline client-sdk/ts-web/rt/src/signature_secp256k1.ts:44]: wait why is this space gone

**Comment by pro-wh** [comment]
rebased, bumped versions

**Comment by pro-wh** [comment]
gonna see about using `@noble/curves` for ed25519 as well

**Comment by pro-wh** [comment]
opening this up for review, without any tweetnacl removal. I looked and I ended up getting all paranoid about ADR-0009

major change: EllipticSigner from client-rt is replaced with NobleSigner, with a similar interface. it's not tested against anything external though, we only check that the sdk can verify its own signature

**Comment by pro-wh** [comment]
ok good, secp256k1 NobleSigner matches up with the runtime-sdk rust code

```ts
import * as oasis from '@oasisprotocol/client';
import * as oasisRT from '@oasisprotocol/client-rt';

describe('secp256k1', () => {
    it('is compatible with runtime-sdk', async () => {
        // Based on `test_memory_signer` in
        // runtime-sdk/src/crypto/signature/mod.rs.
        const priv = await oasis.hash.hash(oasis.misc.fromString('memory signer test'));
        expect(oasis.misc.toHex(priv)).toEqual('4844bcd03d683656f9b7b656f5a53883f7aad7a74d8781fd953d77b5e59956de');
        const signer = oasisRT.signatureSecp256k1.NobleSigner.fromPrivate(priv, 'this key is not important');
        const contextSigner = new oasisRT.signatureSecp256k1.BlindContextSigner(signer);
        const pk = contextSigner.public();
        expect(oasis.misc.toHex(pk)).toEqual('0305327494815250eb0850beb36e78c512a1a42d37503f18843d5eb55554742117');
        const ctx = 'oasis-core/test: context';
        const message = oasis.misc.fromString('this is a message');
        const signature = await contextSigner.sign(ctx, message);
        expect(oasis.misc.toHex(signature)).toEqual('304502210080e4f0e7a94baea44dcded16fb543642291aad5d7ab77606f4fce4cfc5b119260220010c0b40a09803494d7339bb52fb044028c9a58b0964610d863a50274264bd18');
        expect(await oasisRT.signatureSecp256k1.verify(ctx, message, signature, pk)).toEqual(true);
    });
});
```

gotta sort out the global `crypto.getRandomValues` in jest

**Comment by pro-wh** [comment]
adding this to the test :upside_down_face: 

```js
import {webcrypto} from 'crypto';

if (typeof crypto === 'undefined') {
    // @ts-expect-error there are some inconsequential type differences
    globalThis.crypto = webcrypto;
}
```

**Comment by pro-wh** [comment]
> other projects using Node will have to do similar patches?

ts-web targets web only, it's based on grpc-web, which internally uses XMLHttpRequest

I think our last recommendation was to use `--experimental-global-webcrypto`. see more hacks here https://github.com/oasisprotocol/oasis-sdk/blob/main/client-sdk/ts-web/core/playground/e2e-test-nodejs.js

**Comment by pro-wh** [comment]
web wallet builds successfully with this version and minor change:

```diff
diff --git a/src/app/lib/transaction.ts b/src/app/lib/transaction.ts
index ff834281..c46f2cc1 100644
--- a/src/app/lib/transaction.ts
+++ b/src/app/lib/transaction.ts
@@ -19,7 +19,7 @@ export const signerFromPrivateKey = (privateKey: Uint8Array) => {
 }
 
 export const signerFromEthPrivateKey = (ethPrivateKey: Uint8Array) => {
-  return oasisRT.signatureSecp256k1.EllipticSigner.fromPrivate(ethPrivateKey, 'this key is not important')
+  return oasisRT.signatureSecp256k1.NobleSigner.fromPrivate(ethPrivateKey, 'this key is not important')
 }
 
 /** Transaction Wrapper */
```

**Pull request merged**